File: misanuncios-ui/src/app/infrastructure/dto/message.dto.ts
import { ConversationDto } from "./conversation.dto";
import { UserDto } from "./user.dto";

export interface MessageDto
{
  id: number;
  sender: UserDto | null | number; // Assuming User model exists
  message: string;
  sendAt: Date;
  isRead: boolean;
  conversation: ConversationDto | null; // Assuming Conversation model exists


}

File: misanuncios/src/main/java/com/yaagoub/misanuncios/infrastructure/rest/spring/dto/MessageDto.java
package com.yaagoub.misanuncios.infrastructure.rest.spring.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import com.yaagoub.misanuncios.infrastructure.db.database.model.ConversationEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

import java.util.Date;
@Data
@EqualsAndHashCode(exclude = {"sender", "conversation"})

public class MessageDto {

    private long id;
    private UserDto sender;
    private String message;
    private Date sendAt;
    private boolean isRead;
    @JsonIgnore
    private ConversationDto conversation;

}
